state/mysql/mysql.go

```diff
@@ -305,6 +371,44 @@ func (m *MySQL) ensureStateTable(ctx context.Context, stateTableName string) err
 		}
 	}
 
+	// Check if expiredate column exists - to cater cases when table was created before v1.11.
+	columnExists, err := columnExists(ctx, m.db, schemaName, stateTableName, "expiredate", m.timeout)
+	if err != nil {
+		return err
+	}
+
+	if !columnExists {
+		m.logger.Infof("Adding expiredate column to MySql state table '%s'", stateTableName)
+		_, err = m.db.ExecContext(ctx, fmt.Sprintf(
+			`ALTER TABLE %s ADD COLUMN expiredate TIMESTAMP NULL;`, stateTableName))
+		if err != nil {
+			return err
+		}
+		_, err = m.db.ExecContext(ctx, fmt.Sprintf(
+			`CREATE INDEX expiredate_idx ON %s (expiredate);`, stateTableName))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *MySQL) ensureMetadataTable(ctx context.Context, schemaName, metaTableName string) error {
+	exists, err := tableExists(ctx, m.db, schemaName, metaTableName, m.timeout)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		m.logger.Info("Creating MySQL metadata table")
+		_, err = m.db.ExecContext(ctx, fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (
+			id VARCHAR(255) NOT NULL PRIMARY KEY, value VARCHAR(255) NOT NULL);`, metaTableName))
```

Comment:
For `value`, consider using `TEXT` since we may need more than 255 chars in the future. (It's probably ok for the key)

Reply:
I have changed that to `TEXT` for Value. But, just wanted to confirm one thing. Currently, we are storing last-cleanup timestamp in the value, do you foresee any particular use case for metadata table here?